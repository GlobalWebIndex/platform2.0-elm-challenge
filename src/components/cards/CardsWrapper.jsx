import React from "react";
import PropTypes from "prop-types";
import { useStyles } from "./CardsWrapper.style";
import Card from "./Card";

function CardsWrapper({ cardItems }) {
  const classes = useStyles();
  return (
    <section className={classes.gridContainer}>
      {cardItems.map((item) => (
        <Card
          key={item.id}
          imageSrc={item?.imageSrc || null}
          imageWidth={item?.imageWidth || null}
          imageHeight={item?.imageHeight || null}
          id={item.id}
          title={item.title}
          cardLink={item.cardLink}
          isInGrid={true}
        />
      ))}
    </section>
  );
}

CardsWrapper.propTypes = {
  cardItems: PropTypes.array,
};

export default CardsWrapper;
