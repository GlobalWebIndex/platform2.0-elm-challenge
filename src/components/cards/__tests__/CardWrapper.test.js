import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import CardsWrapper from "../CardsWrapper";

describe("<CardWrapper/>", () => {
  it("renders", () => {
    const { container } = render(
      <CardsWrapper
        cardItems={[
          {
            id: "abc",
            imageSrc: "https://cat.com",
            imageHeight: 311,
            imageWidth: 111,
          },
        ]}
      />,
      { wrapper: BrowserRouter }
    );

    expect(container).toMatchSnapshot();
  });
});
