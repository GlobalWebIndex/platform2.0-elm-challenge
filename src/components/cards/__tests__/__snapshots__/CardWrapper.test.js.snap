// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardWrapper/> renders 1`] = `
<div>
  <section
    class="gridContainer-0-2-1"
  >
    <div
      class="cardItem-0-2-2 cardItem-d0-0-2-6"
    >
      <img
        alt="id abc"
        class="cardImage-0-2-3"
        data-testid="card-image"
        height="311"
        loading="lazy"
        src="https://cat.com"
        width="111"
      />
    </div>
  </section>
</div>
`;
